distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
@dmcgowan: distribution knowing about the plugin mediatype should be okay. Are you concerned purely because it is in v0 and will be okay to include it in distribution once we have a stable v1 manifest spec? Also, I agree that code separation is ideal but the above workaround to detecting the container image mediatype feels hackish. What if there are other mediatypes that we are not aware of, but qualify as a good container image today. Then this change  will break a lot of users. That's why blacklisting is safe, since it wont break a thing.